src/libraries/Common/src/Interop/Unix/System.Native/Interop.IsMemberOfGroup.cs

```diff
@@ -8,7 +8,26 @@ internal static partial class Interop
 {
     internal static partial class Sys
     {
-        internal static unsafe uint[]? GetGroups()
+        internal static bool IsMemberOfGroup(uint gid)
+        {
+            if (gid == GetEGid())
+            {
+                return true;
+            }
+
+            uint[]? groups = GetGroups();
```

Comment:
Is `GetGroups()` used by any other methods? If not, we could either manually inline it or move the `Array.IndexOf` to it so we not need to allocate an array for the check.